src/truststore/_macos.py

```diff
@@ -370,7 +370,7 @@ def _verify_peercerts_impl(
                 )
                 assert status == 0  # TODO: Check status
 
-                err = ssl.SSLCertVerificationError()
+                err = ssl.SSLCertVerificationError(cf_error_message)
```

Comment:
Minor suggestion, but I wonder if it would help clarify the message if we prefixed the OS message with something like "Error verifying certificate for {hostname} using system trust store: {cf_error_message}" to provide a bit more context.

(Both here and for Windows)